### DISCUSSION TITLE
Force useFetch or useAsyncData to run on server-side

### DISCUSSION THREAD

**Discussion by farahats9** [nuxt/nuxt#21456] [Questions]
I know this probably has been asked a lot but I did try most of the written solutions and it doesn't work.

here is a simple request and I want it to run on the server-side everytime the page is visited
```typescript
<script setup lang="ts">

const config = useRuntimeConfig();
const { data } = await useFetch('/api/endpoint', { baseURL: config.public.API_BASE_URL, server: true })

</script>
```
This fetches data on server-side only if I do a **complete reload** of the page. If the user is browsing to the page via router link it will not trigger a server fetch but will run it on the client-side and the request appears in the network tab in dev tools.

Anyone please help because I am about to give up on using nuxt 3 for my project since 90% of my problems are with the useFetch functions, I am open to alternatives or 3rd party libs if they can do this.

**Comment by farahats9** [comment]
I have used a server-side component to force the fetch to happen on server side using [this example in docs](https://nuxt.com/docs/guide/directory-structure/components#server-components)

Now the problem appears to be fixed.

It would be great if we have a similar thing for composables because I don't need the template, only the script part. If anyone knows a better way let me know.

**Reply by bertyn99** [reply]
its a composable you want or nuxt plugin

**Reply by iBobik** [reply]
It is good for static text, but it is not useable if I want to have some client-side interactions.

**Comment by ichsanputr** [comment]
Anyone know how to fix this?

**Comment by xjccc** [comment]
emm..you could use a `full-path` for `nuxt-link` to load page?

```
<NuxtLink to="https://xxxxx/path">jumpto</NuxtLink>
```

**Comment by Suniron** [comment]
Hey,

The only way I found is this one:

- You need to use `navigateTo` with `external` option set to `true` to refresh all server side data
- You fetch data from the server component  (`*.server.vue`) and store them in `useState('mySharedData', ( ) => dataFromFetchOrAnything` to allow the possibility of retrieving it from the client component (`*. client.vue`) using a `useState` with the same key.

I think (and hope...) there is a largely better way to do this but I didn't find it in the Nuxt documentation 🥹.

There is no information about the using a full server side rendering (to avoid API routes exposition for eg).

There is a subject 😅

**Comment by castrix** [comment]
try to wrap it inside process.server
```
if (process.server) {
server only fetch
}
```

**Comment by Suniron** [comment]
> try to wrap it inside process.server
> ```
> if (process.server) {
> server only fetch
> }
> ```

If the component is re-rendered during the client navigation,  the data are not fetched from the backend with this approach

**Comment by ichsanputr** [comment]
Same here happening to me last weeks when i try to develop my nuxt 3 app [ExcelClash](https://excelclash.com)
